src/Symfony/Component/Ldap/Adapter/ExtLdap/Connection.php

```diff
@@ -85,6 +85,65 @@ public function bind(?string $dn = null, #[\SensitiveParameter] ?string $passwor
         $this->bound = true;
     }
 
+    /**
+     * @param string $password WARNING: When the LDAP server allows unauthenticated binds, a blank $password will always be valid
+     *
+     * @return void
+     */
+    public function sasl_bind(?string $dn = null, #[\SensitiveParameter] ?string $password = null, ?string $mech = null, ?string $realm = null, ?string $authc_id = null, ?string $authz_id = null, ?string $props = null): void
+    {
+        if (!function_exists('ldap_sasl_bind')) {
+            throw new LdapException('Library - missing SASL support');
+        }
+
+        if (!$this->connection) {
+            $this->connect();
+        }
+
+        if (false === @ldap_sasl_bind($this->connection, $dn, $password, $mech, $realm, $authc_id, $authz_id, $props)) {
+            $error = ldap_error($this->connection);
+            switch (ldap_errno($this->connection)) {
+                case self::LDAP_INVALID_CREDENTIALS:
+                    throw new InvalidCredentialsException($error);
+                case self::LDAP_TIMEOUT:
+                    throw new ConnectionTimeoutException($error);
+                case self::LDAP_ALREADY_EXISTS:
+                    throw new AlreadyExistsException($error);
+            }
+            throw new ConnectionException($error);
+        }
+
+        $this->bound = true;
+    }
+
+
+    /**
+     * ldap_exop_whoami accessor, returns authenticated DN
+     *
+     * @return string
+     */
+    public function whoami() : string
+    {
+        $authz_id = ldap_exop_whoami($this->connection);
+        if ($authz_id === false) {
```

Comment:
would ldap_error work here?
```suggestion
        if (false === $authzId = ldap_exop_whoami($this->connection)) {
            throw new LdapException(ldap_error($this->connection));
```